testbench/rest_server.py

```diff
@@ -901,9 +901,15 @@ def handle_gzip_compressed_request():
 @upload.route("/b/<bucket_name>/o", methods=["POST"])
 @retry_test(method="storage.objects.insert")
 def object_insert(bucket_name):
+    # GCS supports "POST" calls for uploading data with an upload_id
```

Comment:
This seems like a good change, but unrelated to this PR?  Or maybe the PR description needs to be more general? e.g. improve resumable uploads?

Reply:
It also looks like this code is tested, but not by a new explicit test, I wonder how that worked before.

Reply:
Good call out. Splitting this into a separate PR #404. Not sure how test coverage was done here 🤷🏻‍♀️ but updated tests for the new changes.